src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -538,8 +540,119 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		if (this.inErrorState) {
 			return undefined;
 		}
+
 		let cell = this.createCell(cellType);
+		return this.insertCell(cell, index);
+	}
+
+	public splitCell(cellType: CellType, index?: number, notebookService?: INotebookService): ICellModel | undefined {
+		if (this.inErrorState) {
+			return undefined;
+		}
+
+		let notebookEditor = notebookService.findNotebookEditor(this.notebookUri);
```

Comment:
You have notebookService as an optional param but then use it immediately here. Is notebookService something that would ever actually be undefined?

Reply:
Fixed